### ISSUE TITLE
Is it possible to set or override the inventory selection using Survey variables?

### ISSUE THREAD

**Issue by MarkLFT** [ansible-semaphore/semaphore#1354]
We have a number of inventory lists, each with several servers, a very common configuration I am sure.  But there are times we want to run a task against a single node, maybe a reboot etc.  

Whilst I can easily override the host / group name using a survey variable, I still end up with needing to create a reboot task template for each inventory list.  Is it possible to override the inventory selection using a survey variable or some other way at runtime?

**Comment by ansibleguy** [comment]
Yes - you can supply the limit inside the task or task-template JSON field:

```json
["--limit", "host1,host2"]
```

Also: I opened a specific feature-request for adding a `--limit` field: https://github.com/ansible-semaphore/semaphore/issues/1357

Regarding multiple inventories - you could try to also supply them using the 'JSON' field as those variables seem to be directly passed to Ansible.

**Comment by Ye-Min-Tun** [comment]
Yes, you can do with survery variable. And update the host variable in your playbook like the following: 
 host: "{{ your-survery-var-name }}" 
This will prompt to set the variable when you run the task. You can enter 'all' or group name or host name. @MarkLFT

**Comment by MarkLFT** [comment]
@Ye-Min-Tun The question was more about if we can select a different inventory file using a survey variable.

**Comment by decanthus314** [comment]
I second this. I am in a situation where it would be very helpful to have a selectable inventory at runtime. 
Preferably a drop drop selector of existing inventories in the semaphore database instead of a survey variable string.

**Comment by ansibleguy** [comment]
That way one should be able to 'override' the inventory-file configured in the 'task-template' when running a 'task' - that's the idea? Or did I get it wrong?
One should currently be able to supply multiple inventory lists using the JSON-field:
```json
["--inventory", "inv2", "--inventory", "inv3"]
```
They should be passed 1-to-1 to the ansible-playbook execution

**Comment by Ye-Min-Tun** [comment]
> @Ye-Min-Tun The question was more about if we can select a different inventory file using a survey variable.

Not sure about selecting the inventory file . 
I just mentioned about selecting individual host or groups of hosts in a single inventory file at runtime.

**Comment by josefglatz** [comment]
> Yes, you can do with survery variable. And update the host variable in your playbook like the following: host: "{{ your-survery-var-name }}" This will prompt to set the variable when you run the task. You can enter 'all' or group name or host name. @MarkLFT

Is it possible that variables can only by have underscores and no dashes? Because this just works when the variable has no dash in it.

**Comment by ansibleguy** [comment]
@josefglatz See [Ansible Documentation for variables](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html#creating-valid-variable-names)
> A variable name can only include letters, numbers, and underscores. [Python keywords](https://docs.python.org/3/reference/lexical_analysis.html#keywords) or [playbook keywords](https://docs.ansible.com/ansible/latest/reference_appendices/playbooks_keywords.html#playbook-keywords) are not valid variable names
